File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area = 0
        left: int = 0
        n: int = len(height)
        right: int = n - 1

        while left < right:
            length: int = min(height[left], height[right])
            breadth: int = right - left
            max_area = max(max_area, length * breadth)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_area
            

        

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {

        int max_water = 0;
        
        int left = 0;
        int n = height.length;
        int right = n - 1;

        while (left < right) {
            int water = (right - left) * Math.min(height[left], height[right]);
            max_water = Math.max(water, max_water);
            if (height[left] < height[right]) {
                left += 1;
            } else {
                right -= 1;
            }
        }

        return max_water;
        
    }
}